### PR TITLE
Number/Currency field enhancements

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Related: #4207 

Similar to what we've done with the date/time fields.

- [x] Late static binding
- [x] Currency field extends Number field (remove duplicate code)
- [x] Put thousands/dot formatting logic in separate method and re-use it
- [x] `value()` method to prevent returning values that should be empty
- [x] Also refactors old `pods_var()` calls in the number field
- [ ] Test and review
- [ ] Convert file indentation to Tabs once PR is ready @JoryHogeveen

### CODE REVIEW

File: classes/fields/number.php [hunk 1]
```diff
@@ -137,68 +137,76 @@ public function options () {
...
+	 * @param array $options
+	 *
+	 * @return string
+	 * @since 2.0
+	 */
+	public function schema( $options = null ) {
 
-        if ( $length < 1 || 64 < $length )
-            $length = 64;
+		$length = (int) pods_v( static::$type . '_max_length', $options, 12, true );
```
Comment (sc0ttkclark) [inline classes/fields/number.php]: Thank you for the pods_v stuff here, it helps.

We call pods_var in so many places, it has an extra sanitization call that's not needed in many cases. So converting to pods_v will help us long term too.

**Comment by sc0ttkclark** [comment]
> Store handling (pre_save()) for empty values. Return 0.00 or null/''?

If we break backcompat, we don't want that. What do you see here for potential backcompat issues?

**Comment by JoryHogeveen** [comment]
> What do you see here for potential backcompat issues?

The fix is already in `value()` so even existing "empty-but-non-empty" values should return an empty string with that fix.
As for backwards compat. I'n not sure how ACT's handle this kind of data. I know we can safely store an empty string in metadata, but what about ACT's tables with fixed data types?

**Comment by sc0ttkclark** [comment]
ACTs / table-based pods will all store the values as 0 or 0.00 or whatever the decimal format is for the table column.

**Comment by jimtrue** [comment]
Number fields, 0, is being recognized as empty by `[if number_field][/if]`: Yay!
Currency fields, 0.00, are still not recognized as empty by `[if currency_field][/if]`: Sad.

**Comment by JoryHogeveen** [comment]
@jimtrue 
Since the results are different. What happens if the Number field decimals setting is set to more than 0 (without soft formating enabled)

**Comment by jimtrue** [comment]
@JoryHogeveen Once you add decimals, the number no longer represents as empty. As noted, this might be in the `[if field]` handling and might have nothing to do with the number itself, if it passes 'empty' check on 0.00

**Pull request merged**